This patch applies to the MH 6.7 version of folder.c
(The line numbers for 6.8.1 are 531,536 => 531,541)
*** folder.c.orig	Fri Oct 22 16:45:10 1993
--- folder.c	Fri Oct 22 16:48:54 1993
***************
*** 521,526 ****
--- 521,531 ----
      *cp = NULL;
  
      base = strcmp (name, "./") ? name : name + 2;/* hack */
+ 
+      /* short-cut to see if directory has any sub-directories */
+ 
+      if (stat (name, &st) != NOTOK && st.st_nlink == 2)
+ 	return;
  
      if ((dd = opendir (name)) == NULL) {
  	admonish (name, "unable to read directory ");
